src/vue/utils.ts

```diff
@@ -31,18 +32,27 @@ export function parseQueryArgs<
   arg2: QueryFunction<TQueryFnData, TQueryKey> | TOptions = {} as TOptions,
   arg3: TOptions = {} as TOptions
 ): TOptions {
-  if (!isQueryKey(arg1)) {
-    return arg1;
-  }
+  let options;
 
-  if (typeof arg2 === "function") {
-    return Object.assign(arg3, {
+  if (!isQueryKey(arg1)) {
+    // `useQuery(optionsObj)`
+    options = arg1;
+  } else if (typeof arg2 === "function") {
+    // `useQuery(queryKey, queryFn, optionsObj?)`
+    options = {
+      ...toRefs(reactive(arg3 as unknown as object)),
       queryKey: arg1,
       queryFn: arg2,
-    });
+    };
+  } else {
+    // `useQuery(queryKey, optionsObj?)`
+    options = {
+      ...toRefs(reactive(arg2 as unknown as object)),
+      queryKey: arg1,
+    };
   }
 
-  return Object.assign(arg2, { queryKey: arg1 });
+  return reactive(options as object) as unknown as TOptions;
```

Comment:
I didn't like the idea of changing (extending) original arguments.
I thought it's better to create a new Options object out of the arguments.
By making the Options object reactive we ensure that all the passed refs are unwrapped

Reply:
I wanted to do this originally, but since i failed to make it work without `reactive` wrapping, i could not spread the object without loosing reactivity. 🎉

Reply:
yeah, I guess so :) 
It also took me time to understand how to make it all possible :)

Though I'm not really satisfied with the current approach. Too much non-transparent data massaging. And it doesn't cover a case with arrays of refs somewhere deep in the Options.

I have an idea to make it better. I probably will create a following-up PR